File: lib/services/inventory_service.dart
import '../models/inventory_item.dart';
import 'firebase_service.dart';

class InventoryService {
  final FirebaseService _firebase;
  final String _path = 'inventory';

  InventoryService(this._firebase);

  Future<String> create(InventoryItem item) async {
    return await _firebase.save(_path, item.toJson());
  }

  Future<void> update(InventoryItem item) async {
    if (item.id == null) throw Exception('Item ID is required');
    await _firebase.update(_path, item.id!, item.toJson());
  }

  Future<void> delete(String id) async {
    await _firebase.delete(_path, id);
  }

  Future<InventoryItem?> get(String id) async {
    final data = await _firebase.get(_path, id);
    return data != null ? InventoryItem.fromJson(data, id) : null;
  }

  Future<List<InventoryItem>> getAll() async {
    final data = await _firebase.getAll(_path);
    return data.map((item) => InventoryItem.fromJson(item, item['id'])).toList();
  }

  Future<List<InventoryItem>> getLowStockItems() async {
    final items = await getAll();
    return items.where((item) => item.isLowStock).toList();
  }
}

File: js/services/inventory-service.js
import { database } from '../firebase-config.js';
import { ref, push, set, get, remove } from 'firebase/database';

export class InventoryService {
    constructor() {
        this.path = 'inventory';
    }

    async create(item) {
        const itemRef = ref(database, this.path);
        const newRef = push(itemRef);
        await set(newRef, item);
        return newRef.key;
    }

    async update(id, item) {
        const itemRef = ref(database, `${this.path}/${id}`);
        await set(itemRef, item);
    }

    async delete(id) {
        const itemRef = ref(database, `${this.path}/${id}`);
        await remove(itemRef);
    }

    async get(id) {
        const itemRef = ref(database, `${this.path}/${id}`);
        const snapshot = await get(itemRef);
        if (snapshot.exists()) {
            return { id: snapshot.key, ...snapshot.val() };
        }
        return null;
    }

    async getAll() {
        const itemsRef = ref(database, this.path);
        const snapshot = await get(itemsRef);
        const items = [];
        if (snapshot.exists()) {
            snapshot.forEach(child => {
                items.push({ id: child.key, ...child.val() });
            });
        }
        return items;
    }

    async getLowStockItems() {
        const items = await this.getAll();
        return items.filter(item => {
            const quantity = item.quantity || 0;
            const minStock = item.minStockLevel || 0;
            return quantity <= minStock;
        });
    }
}

export const inventoryService = new InventoryService();




